code/modules/ruins/lavalandruin_code/elephantgraveyard.dm

```diff
@@ -0,0 +1,227 @@
+
+/obj/structure/sink/oil_well/deconstruct(disassembled = TRUE)
+	qdel(src)
+
+//***Grave mounds.
+/obj/structure/closet/crate/grave
+	name = "burial mound"
+	desc = "An marked patch of soil, showing signs of a burial long ago. You wouldn't disturb a grave... right?"
+	icon = 'icons/obj/crates.dmi'
+	icon_state = "grave"
+	dense_when_open = TRUE
+	material_drop = /obj/item/stack/ore/glass/basalt
+	material_drop_amount = 5
+	anchorable = FALSE
+	anchored = TRUE
+	locked = TRUE
+	breakout_time = 900
+	cutting_tool = /obj/item/shovel
+	var/lead_tomb = FALSE
+	var/first_open = FALSE
+	var/untouched = TRUE
+
+/obj/structure/closet/crate/grave/PopulateContents()  //GRAVEROBBING IS NOW A FEATURE
+	..()
+	switch(rand(1,8))
+		if(1)
+			new /obj/item/coin/gold(src)
+			new /obj/item/storage/wallet(src)
+		if(2)
+			new /obj/item/clothing/glasses/meson(src)
+		if(3)
+			new /obj/item/coin/silver(src)
+			new /obj/item/shovel/spade(src)
+		if(4)
+			new /obj/item/storage/book/bible/booze(src)
+		if(5)
+			new /obj/item/clothing/neck/stethoscope(src)
+			new	/obj/item/scalpel(src)
+			new /obj/item/hemostat(src)
+
+		if(6)
+			new /obj/item/reagent_containers/glass/beaker(src)
+			new /obj/item/clothing/glasses/science(src)
+		if(7)
+			new /obj/item/clothing/glasses/sunglasses(src)
+			new /obj/item/clothing/mask/cigarette/rollie(src)
+
+/obj/structure/closet/crate/grave/open(mob/living/user, obj/item/S)
+	if(opened == FALSE)
+		to_chat(user, "<span class='notice'>The ground here is too hard to dig up with your bare hands. You'll need a shovel.")
+	else
+		to_chat(user, "<span class='notice'>The grave has already been dug up.</span>")
+
+/obj/structure/closet/crate/grave/tool_interact(obj/item/S, mob/living/carbon/user)
+	if(user.a_intent == INTENT_HELP) //checks to attempt to dig the grave, must be done on help intent only.
+		if(opened == FALSE)
+			if(istype(S,cutting_tool) && S.tool_behaviour == TOOL_SHOVEL)
+				to_chat(user, "<span class='notice'>You start start to dig open \the [src]  with \the [S]...</span>")
+				if (do_after(user,20, target = src))
+					opened = TRUE
+					locked = TRUE
+					dump_contents()
+					update_icon()
+					SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "graverobbing", /datum/mood_event/graverobbing)
+					if(lead_tomb == TRUE && first_open == TRUE)
+						user.gain_trauma(/datum/brain_trauma/magic/stalker)
+						to_chat(user, "<span class='boldwarning'>Oh no, no no no, THEY'RE EVERYWHERE! EVERY ONE OF THEM IS EVERYWHERE!</span>")
+						first_open = FALSE
+					if(untouched == TRUE)
+						new /obj/effect/decal/remains/human(loc)
+						untouched = FALSE
+					return 1
+				return 1
+			else
+				to_chat(user, "<span class='notice'>You can't dig up a grave with \the [S.name].</span>")
+				return 1
+		else
+			to_chat(user, "<span class='notice'>The grave has already been dug up.</span>")
+			return 1
+
+	else if(user.a_intent != INTENT_HELP) //checks to attempt to remove the grave entirely.
+		if(opened == TRUE)
```

Comment:
!opened

Reply:
My intent here was so that you can only fully remove graves after they've been dug up , but checking if(opened) still seems to work here.

Reply:
oh yeah, that would be correct. my bad